testMemoryCompiler.py

```diff
@@ -0,0 +1,182 @@
+# This is work in progress code for allowing extensions of java classes from within
+# jpype.
+#
+# Theory of operation:
+# - The jvm must be started so that we have access to reflection.
+# - We have a class which is a mock up for the extension class with 
+#   annotations for the class we are extending and for each method
+#   we wish to override.
+# - When the class is encountered, it writes a section of java code
+#   which creates a new class for which the methods are pointed to 
+#   an invocation handler.
+# - The python class creates a special invocation handler similar to the 
+#   proxy which is static in the class and points to python class.
+# - When the new class is constucted from within python, the class will get
+#   a pointer to the python object and a reference to hold scope.
+#
+# Issues:
+# - Generating a code segment to cover all the cases will be difficult.
+# - Exceptions need to be passed through, thus constructors with exceptions
+#   require complete specification.
+# - Extending a generic will be challenging as reflection does not provide
+#   everything in terms of names at least in some versions.
+# - Writting code stubs that apply for many versions of java may be an issue.
+#   I doubt that I can make it work for 1.6.
```

Comment:
Is java6 still being used a lot? We can think of cutting old pig tails then.

Reply:
Not as far as I am aware.  I am a bit worried about the effects of Java 9 and the changes to generics.